### PR TITLE
Convert EMWF to use paginate_options

### PR DESCRIPTION
**Pull request by esoergel**
There was a bunch of repetitive index manipulation that was happening, and it was pretty difficult to modify or verify that it worked properly, so this PR pulls out the pagination logic and tests that, then moves stuff around a bunch to use that function.  This shouldn't have any user-facing changes.
This came out of http://manage.dimagi.com/default.asp?175315, and it's the cleanup I promised in https://github.com/dimagi/commcare-hq/pull/7337
I spoke with @dannyroberts about how to do this, and @nickpell wrote the `paginate_options` function with me.
@sravfeyn
Happy to walk someone through this if needed

### CODE REVIEW

File: corehq/apps/reports/filters/api.py [hunk 1]
```diff
@@ -73,81 +54,88 @@ def get(self, request, domain):
...
+
+    def get_static_options_size(self, query):
+        return len(self.get_all_static_options(query))
+
+    def get_all_static_options(self, query):
+        return [user_type for user_type in self.utils.static_options
+                if query.lower() in user_type[1].lower()]
+
+    def get_static_options(self, query, start, size):
+        return self.get_all_static_options(query)[start:start+size]
```
Comment (dimagimon) [inline corehq/apps/reports/filters/api.py]: E226 missing whitespace around arithmetic operator

File: corehq/apps/reports/tests/test_filters.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        def matching_objects(query):
+            if not query:
+                return options
+            return [o for o in options if query.lower() in o.lower()]
+
+        def get_size(query):
+            return len(matching_objects(query))
+
+        def get_objects(query, start, size):
+            return matching_objects(query)[start:start+size]
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_filters.py]: E226 missing whitespace around arithmetic operator

File: corehq/apps/reports/filters/api.py [hunk 3]
```diff
@@ -31,29 +30,11 @@ def utils(self):
...
-                    "query": q,
-                    "fields": ["first_name", "last_name", "username"],
-                }} for q in queries
-            ]}}
-            self.group_query = {"bool": {"must": [
-                {"query_string": {
-                    "query": q,
-                    "default_field": "name",
-                }} for q in queries
-            ]}}
```
Comment (esoergel) [inline corehq/apps/reports/filters/api.py]: These were moved into their respective queries to depend less on object state

File: corehq/apps/reports/filters/api.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/reports/filters/api.py]: do you guys agree with this?  Which looks best?
```python
[start:start+size]
[start:start + size]
[start : start + size]
```

Comment (sravfeyn) [inline corehq/apps/reports/filters/api.py]: One you have here looks good to me

**Comment by sravfeyn** [comment]
Kicked build again. Looks good. Will merge once the build passes

**Comment by czue** [comment]
@esoergel haven't reviewed yet, but this category of change often triggers my "something is going to break" radar, so just wondering how you feel about QA? I see you added tests, though I find that reports and report filters have lots of edge cases

**Comment by esoergel** [comment]
@czue I'm in no rush to get this through, although I tested it pretty extensively locally (several domains, both versions of the filter, making sure I hit all the data sources).  Amelia has some related changes in mind for me (seeing if we can conditionally display some of the static options), so I can hold off on this and QA/merge both of those together.

**Comment by czue** [comment]
I wasn't saying we need to do QA, just wanted to make sure the question was asked. :)

**Comment by esoergel** [comment]
It seems fine to wait on the merge, as long as no one else is going to touch this code in the meantime.

**Comment by czue** [comment]
Famous last words...

**Comment by sravfeyn** [comment]
Looks good to me. Are we waiting on QA?

**Comment by esoergel** [comment]
I was gonna hold off on a second related feature, but that actually looks farther away than I expected, so I'd like to get this in sooner.  Can you merge whenever you're satisfied?  Everything but the last commit is on staging if you wanna click around:
https://staging.commcarehq.org/a/testclark/reports/case_list/
https://staging.commcarehq.org/a/testclark/reports/daily_form_stats/

**Comment by sravfeyn** [comment]
Cool, will merge after the build finishes. If not feel free to self-merge.

**Pull request merged**